src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java

```diff
@@ -0,0 +1,248 @@
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+        };
+        final Object res = this.runCommandLine(args);
+        final File tribbleIndex = Tribble.indexFile(ORIG_FILE);
+        Assert.assertEquals(res, tribbleIndex.getAbsolutePath());
+        tribbleIndex.deleteOnExit();
+
+        final Index index = IndexFactory.loadIndex(res.toString());
+        Assert.assertTrue(index instanceof LinearIndex);
+        for (int chr = 1; chr <= 4; chr++) {  //note: unusual loop
+            Assert.assertTrue(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (int chr = 5; chr <= 22; chr++) { //note: unusual loop
+            Assert.assertFalse(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (final String chr : Arrays.asList("X", "Y", "MT")){
+            Assert.assertFalse(index.containsChromosome(chr));
+        }
+
+        Assert.assertEquals(index.getSequenceNames(), Arrays.asList("1", "2", "3", "4"));
+    }
+
+    @Test
+    public void testVCFGZIndex() {
+        final File ORIG_FILE = new File(getTestDataDir(), "test_variants_for_index.vcf.gz");
+        final File outName = BaseTest.createTempFile("test_variants_for_index.", ".idx");
+
+        final String[] args = new String[]{
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+                "-O" ,  outName.getAbsolutePath()
+        };
+        final Object res = this.runCommandLine(args);
+        Assert.assertEquals(res, outName.getAbsolutePath());
+
+        final Index index = IndexFactory.loadIndex(res.toString());
+        Assert.assertTrue(index instanceof TabixIndex);
+        for (int chr = 1; chr <= 4; chr++) {  //note: unusual loop
+            Assert.assertTrue(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (int chr = 5; chr <= 22; chr++) { //note: unusual loop
+            Assert.assertFalse(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (final String chr : Arrays.asList("X", "Y", "MT")){
+            Assert.assertFalse(index.containsChromosome(chr));
+        }
+
+        Assert.assertEquals(index.getSequenceNames(), Arrays.asList("1", "2", "3", "4"));
+    }
+
+    @Test(enabled = false)
+    public void testVCFGZIndex_inferredName() throws IOException {
+        final File ORIG_FILE = new File(getTestDataDir(), "test_variants_for_index.vcf.gz");
+        final String[] args = new String[]{
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+        };
+        final Object res = this.runCommandLine(args);
+        final File tabixIndex = new File(ORIG_FILE.getAbsolutePath() + IndexFeatureFile.TABIX_INDEX_EXTENSION);;
+        Assert.assertEquals(res, tabixIndex.getAbsolutePath());
+        tabixIndex.deleteOnExit();
+
+
+        final Index index0 = IndexFactory.loadIndex(res.toString()); //BUG <- this blows up
+        final Index index1 = new TabixIndex(tabixIndex);             //BUG <- this blows up
+    }
+
```

Comment:
Add a test with a non-VCF/BCF `.gz` file, and show that you get a `UserException` as expected (since the tool only supports indexing of block-compressed files when they are in VCF format). Also, what happens with a `.bcf.gz`? BCF is already compressed, so this should probably produce an error as well.